cpp/src/column/column_device_view.cu

```diff
@@ -36,77 +36,74 @@ void column_device_view::destroy() {
   delete this;
 }
 
-// Place any child objects in host memory (h_ptr) and use the device
-// memory ptr (d_ptr) to set any child object pointers.
-column_device_view::column_device_view( column_view source, void * h_ptr, void* d_ptr )
-    : detail::column_device_view_base{source.type(),       source.size(),
-                                      source.head(),       source.null_mask(),
-                                      source.null_count(), source.offset()},
-      _num_children{source.num_children()}
+namespace
 {
+
+// helper function for use by column_device_view and mutable_column_device_view constructors
+template<typename ColumnView, typename ColumnDeviceView>
+ColumnDeviceView* child_columns_to_device_array( ColumnView const& source, void* h_ptr, void* d_ptr )
+{
```

Comment:
I'd like to see a more detailed description of what this does as the comment.   It's doing more than returning an array of devices / device memory.  It's initializing host objects as well and doing the whole thing recursively.    Might benefit from a more descriptive name as well.

Also, see comments below about the relationship of this function to the constructor and the overall class.